File: 682-baseball-game/baseball-game.java
class Solution {
    public int calPoints(String[] operations) {
        Stack<Integer> records = new Stack<>();
        for(String s : operations){
            if(s.equals("C")){
                records.pop();
            }else if(s.equals("D")){
                int temp = records.peek();
                int new_record  = temp*2;
                records.push(new_record);
            }else if(s.equals("+")){
                int one =0 ;
                int two = 0 ;
                int new_record = 0;
                one  = records.peek();
                records.pop();
                two =records.peek();
                records.pop();
                new_record = one + two;
                records.push(two);
                records.push(one);
                records.push(new_record);
            }else{
                int temp = Integer.parseInt(s);
                records.push(temp);

            }
        }
        int sum =0;
        while(!records.empty()){
            sum += records.peek();
            records.pop();
        }
        return sum;
    }
}

File: 682-baseball-game/baseball-game.cpp
class Solution {
public:
    int calPoints(vector<string>& operations) {
        stack<string> st;
        for(string s : operations){
            if(s == "+"){
                int temp1 = stoi(st.top());
                st.pop();
                int temp2 = stoi(st.top());
                st.pop();
                st.push(to_string(temp2));
                st.push(to_string(temp1));
                st.push(to_string(temp1 + temp2));
            }else if(s == "D"){
                string temp = st.top();
                st.push(to_string(2*stoi(temp)));
            }else if(s == "C"){
                st.pop();
            }
            else{
                st.push(s);
            }
        }
        int count =0 ;
        while(!st.empty()){
            count = count + stoi(st.top());
            st.pop();
        }
        return count;
    }
};